libsolidity/parsing/Parser.cpp

```diff
@@ -1299,7 +1299,7 @@ ASTPointer<InlineAssembly> Parser::parseInlineAssembly(ASTPointer<ASTString> con
 	if (block == nullptr)
 		BOOST_THROW_EXCEPTION(FatalError());
 
-	location.end = block->location.end;
+	location.end = block->debugData->location.end;
```

Comment:
I wonder though why we need to use the location from block, and not the current position we have here?

Reply:
Because the current position is usually already the next token. There is also a difference between the end of one token and the start of the next (i.e. the whitespace in between).